x-pack/plugins/ml/public/embeddables/anomaly_charts/anomaly_charts_embeddable_factory.tsx

```diff
@@ -0,0 +1,232 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { i18n } from '@kbn/i18n';
+import { KibanaRenderContextProvider } from '@kbn/react-kibana-context-render';
+import React, { useMemo } from 'react';
+import type { StartServicesAccessor } from '@kbn/core/public';
+import type { Observable } from 'rxjs';
+import { Subscription, combineLatest, map, skipWhile } from 'rxjs';
+import type { PublishingSubject } from '@kbn/presentation-publishing';
+import { apiHasParentApi, apiPublishesTimeRange, fetch$ } from '@kbn/presentation-publishing';
+import useUnmount from 'react-use/lib/useUnmount';
+import { KibanaContextProvider } from '@kbn/kibana-react-plugin/public';
+import {
+  apiHasExecutionContext,
+  initializeTimeRange,
+  initializeTitles,
+} from '@kbn/presentation-publishing';
+import useObservable from 'react-use/lib/useObservable';
+import { of } from 'rxjs';
+import type { ReactEmbeddableFactory } from '@kbn/embeddable-plugin/public';
+import type { TimeRange } from '@kbn/es-query';
+import { css } from '@emotion/react';
+import type { MlPluginStart, MlStartDependencies } from '../../plugin';
+import type { AnomalyChartsEmbeddableApi, AnomalyChartsEmbeddableState } from '..';
+import { ANOMALY_EXPLORER_CHARTS_EMBEDDABLE_TYPE } from '..';
+import { useReactEmbeddableExecutionContext } from '../common/use_embeddable_execution_context';
+import { initializeAnomalyChartsControls } from './initialize_anomaly_charts_controls';
+import { LazyEmbeddableAnomalyChartsContainer } from './lazy_anomaly_charts_container';
+import { getAnomalyChartsServiceDependencies } from './get_anomaly_charts_services_dependencies';
+import { buildDataViewPublishingApi } from '../common/anomaly_detection_embeddable';
+
+export const getAnomalyChartsReactEmbeddableFactory = (
+  getStartServices: StartServicesAccessor<MlStartDependencies, MlPluginStart>
+) => {
+  const factory: ReactEmbeddableFactory<
+    AnomalyChartsEmbeddableState,
+    AnomalyChartsEmbeddableApi,
+    AnomalyChartsEmbeddableState
```

Comment:
nit: You don't need this third type argument. The runtime state type defaults to the same as the serialized state.

Reply:
Thanks for the hint. Updated here https://github.com/elastic/kibana/pull/183456/commits/1ed16a4c27a7d094f8c60beda5f13c7b95d58c7a